### PR TITLE
coverage: line 74 of tree.py file, creates tree with TreeSource data

### PR DESCRIPTION
**Pull request by renantamashiro**
## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

**Comment by renantamashiro** [comment]
Maybe I need to test whether each Node of TreeSource can have children, like this:

`tree = TreeSource(data = {('first', 1): None, ('second', 2): [], accessors=''}`
`self.assertFalse(tree.data[0].can_have_children()) -> because is None`
`self.assertTrue(tree.data[1].can_have_children()} -> because is a List`

That's right?

**Comment by freakboy3742** [comment]
You don't need to check the behavior of TreeSource - you're looking to confirm that the tree is using the TreeSource correctly. That means confirming:
 * Is the data actually a tree source?
 * Is it a tree source that is consistent with the data that was originally provided (i.e., not only is data a valid TreeSource, but it's the *same* tree source as the one that was provided)?

Confirm *those* properties, and you'll have a solid test.

File: src/core/tests/widgets/test_tree.py [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import toga
 import toga_dummy
 from toga_dummy.utils import TestCase
-from toga.sources import TreeSource, Source
+from toga.sources import TreeSource
```
Comment (renantamashiro) [inline src/core/tests/widgets/test_tree.py:4]: Ignore this... Source imported but unused and I deleted

**Comment by renantamashiro** [comment]
@freakboy3742 I understood your review, thanks 👍

In this commit, I included data and accessors for TreeSource and tested with AssertEqual - using these accessors.

Should I do another test?

**Pull request merged**